packages/config/src/options.js

```diff
@@ -324,5 +324,20 @@ export function getNuxtConfig(_options) {
     bundleRenderer.runInNewContext = options.dev
   }
 
+  // Enable rate limit for dev mode
+  if (options.dev) {
```

Comment:
Any reason for enable it in dev mode ?
I think there is no big value to limit rate in development except testing

Reply:
It's to fix Safari issue, see #3828.

Reply:
@clarkdo the safari bug mentioned in the original text ☺️  (#3828)

Reply:
But I agree that it could create unwanted side effects on development (getting a `429` status code when requesting static files)

Reply:
I'll make cache key `ip + url`. Actually it is an odd behaviour from browser to request same resource more than once.

Reply:
For dev mode, I think ip with a bigger max is better

Reply:
This may cause bad DX. Like display rate limit error page if an internal ajax was repeated.

Reply:
I agree with @clarkdo, but maybe just only enable this middleware by default for safari. As in _don't fix it if it aint broken_ in Linux, Chrome or other browsers. 
The modern middleware already uses ua-parser, but if you think its to cumbersome to check rate-limiting based on the browser then please include a config option that reflects this behaviour is a workaround for an issue with safari and use `os.type() === 'Darwin'` to enable that by default

Reply:
@pimlie This is not just a workaround for Safari bug. Please see PR description.

Reply:
My concern is that if check it with ip and url, it may be not that useful.
If only works for a lot of frequent single resource request.

I suppose the rate-limit may have `per user`/`per service` which has longer time window and bigger maximum count, and `per url` which time window is shorter。

For dev mode, it's not that strict, it's fine for me to use current config for detect some infinite loop or duplicate renderering.